server/configuration-validation.js

```diff
@@ -1,125 +1,192 @@
+      if (configuration !== undefined && !isObject(configuration)) {
+        log.error(`The collection "${name}" should be of type object`)
+      }
 
-  // Validate the `apikey` parameter
-  if (apiKey !== undefined && typeof apiKey !== 'string') {
-    strapi.log.error('`apiKey` should be a string in Meilisearch configuration')
-    delete config.apiKey
-  }
+      return this
+    },
+    validateIndexName() {
+      // indexName is either undefined or a none empty string
+      if (
+        (indexName !== undefined && typeof indexName !== 'string') ||
+        indexName === ''
+      ) {
+        log.error(
+          `the "indexName" option of "${name}" should be a non-empty string`
+        )
+      } else if (indexName !== undefined) {
+        options.indexName = indexName
+      }
+
+      return this
+    },
+    validateTransformEntry() {
+      // transformEntry is either undefined or a function
+      if (
+        transformEntry !== undefined &&
+        typeof transformEntry !== 'function'
+      ) {
+        log.error(
+          `the "transformEntry" option of "${name}" should be a function`
+        )
+      } else if (transformEntry !== undefined) {
+        options.transformEntry = transformEntry
+      }
+
+      return this
+    },
+    validateFilterEntry() {
+      // filterEntry is either undefined or a function
+      if (filterEntry !== undefined && typeof filterEntry !== 'function') {
+        log.error(`the "filterEntry" option of "${name}" should be a function`)
+      } else if (filterEntry !== undefined) {
+        options.filterEntry = filterEntry
+      }
 
-  for (const collection in collections) {
-    config[collection] = validateCollectionConfiguration({
-      configuration: collections[collection],
-      collection: collection,
-    })
+      return this
+    },
+    validateMeilisearchSettings() {
+      // Settings is either undefined or an object
+      if (settings !== undefined && !isObject(settings)) {
+        log.error(`the "settings" option of "${name}" should be an object`)
+      } else if (settings !== undefined) {
+        options.settings = settings
+      }
+
+      return this
+    },
+    validatePopulateEntryRule() {
+      // PopulateEntry is either undefined or an object/array/string`
+      if (
+        populateEntryRule !== undefined &&
+        !isObject(populateEntryRule) &&
+        !Array.isArray(populateEntryRule) &&
+        typeof populateEntryRule !== 'string'
+      ) {
+        log.error(
+          `the "populateEntryRule" option of "${name}" should be an object/array/string`
+        )
+      } else if (populateEntryRule !== undefined) {
+        options.populateEntryRule = populateEntryRule
+      }
+
+      return this
+    },
+    validateNoInvalidKeys() {
+      // Keys that should not be present in the configuration
+      Object.keys(excedent).map(key => {
+        log.warn(`The attribute "${key}" of "${name}" is not a known option`)
```

Comment:
You should check your error/warn messages, it seems some start with an uppercase and other don't